test/subspace-test-service/src/lib.rs

```diff
@@ -528,11 +528,12 @@ impl MockPrimaryNode {
     }
 
     /// Remove a ready transaction from transaction pool.
-    pub fn prune_tx_from_pool(&self, tx: &OpaqueExtrinsic) -> Result<(), Box<dyn Error>> {
+    pub async fn prune_tx_from_pool(&self, tx: &OpaqueExtrinsic) -> Result<(), Box<dyn Error>> {
         self.transaction_pool.pool().prune_known(
             &BlockId::Hash(self.client.info().best_hash),
             &[self.transaction_pool.hash_of(tx)],
         )?;
+        tokio::time::sleep(time::Duration::from_millis(1)).await;
```

Comment:
This makes some sense to me, please add a comment here to explain why we need to wait for 1ms.

Reply:
1ms makes no sense to me. There is no guarantee that this 1ms will have an actual impact on async tasks. This is not JavaScript, runtime is multi-threaded. It will still be flaky.